components/brave_wallet/browser/internal/hd_key_ed25519.h

```diff
@@ -38,8 +38,8 @@ class HDKeyEd25519 : public HDKeyBase {
   // If path contains normal index, nullptr will be returned
   std::unique_ptr<HDKeyBase> DeriveChildFromPath(
       const std::string& path) override;
-  std::vector<uint8_t> Sign(const std::vector<uint8_t>& msg,
-                            int* recid = nullptr) override;
+  std::vector<uint8_t> SignCompact(const std::vector<uint8_t>& msg,
+                                   int* recid = nullptr) override;
```

Comment:
This name doesn't seem right on ed25519, please introduce a new virtual function `virtual std::vector<uint8_t> Sign(const std::vector<uint8_t>& msg)` for it and then make `SolanaKeyring` overrides `SignMessage` and call the new `HDKey::Sign`

Reply:
After some thinking I've removed Sign function from base class at all as 
1) Signing is different for each coin
2) All signing is done not through some reference to base class, but always within some coin's context.
So we have these entry points for signing somethings
```
KeyringService::SignMessageByBitcoinKeyring
KeyringService::SignTransactionByFilecoinKeyring
KeyringService::SignTransactionByDefaultKeyring
KeyringService::SignMessageByDefaultKeyring
KeyringService::SignMessageBySolanaKeyring
```
and each of them accesses exact type of key and sing algorithm.

I'm also have an idea that HDKeyBase should just be able to provide private key to keyrings classes which do all signing themselves, but not worth changing something like that now.